source/AI.cpp

```diff
@@ -3359,32 +3359,118 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 		// Board the nearest disabled ship, focusing on hostiles before allies. Holding
 		// `Shift` results in boarding only player-owned escorts in need of assistance.
 		shared_ptr<const Ship> target = ship.GetTargetShip();
-		if(!target || !CanBoard(ship, *target) || (shift && !target->IsYours()))
+		if(!target || (target && activeCommands.Has(Command::WAIT)) || !CanBoard(ship, *target)
+				|| (shift && !target->IsYours()))
 		{
 			if(shift)
 				ship.SetTargetShip(shared_ptr<Ship>());
 
-			double closest = numeric_limits<double>::infinity();
 			bool foundEnemy = false;
-			bool foundAnything = false;
-			for(const shared_ptr<Ship> &other : ships)
-				if(CanBoard(ship, *other))
+			const auto targetPriority = Preferences::BoardingSetting();
+			const bool costPriority = (targetPriority == "value");
+			const bool distancePriority = (targetPriority == "proximity");
```

Comment:
Do you mean for the return of the function itself? Other preferences use text.

If you mean for the handling with enum instead of bools it'd mean having to check enum == DISTANCE or watever all the time instead of just checking on the bool

Reply:
Use an enum class instead of the books.

I would use a string enum class map to convert the string from preferences to the appropriate enum class, and checking the value of the enum clas type should just involve int comparisons, which should be just as fast as bool comparisons.

I think it'd also make this more maintainable.

Reply:
I never meant it would be a problem for performances but readability altough its been a while since I looked at this code.
Ty for clarifying I'll use an enum class if you prefer it that way, just feels a bit odd to me to do so is all.
Should it be declared as a static const just when we need it in the function?

Reply:
This is what you meant, right?